apex_launchtest/apex_launchtest/apex_runner.py

```diff
@@ -24,79 +24,98 @@
 from launch.event_handlers import OnProcessIO
 
 from .io_handler import ActiveIoHandler
-from .loader import PostShutdownTestLoader, PreShutdownTestLoader
 from .parse_arguments import parse_launch_arguments
 from .proc_info_handler import ActiveProcInfoHandler
 from .test_result import FailResult, TestResult
 
 
-def _normalize_ld(launch_description_fn):
-    # A launch description fn can return just a launch description, or a tuple of
-    # (launch_description, test_context).  This wrapper function normalizes things
-    # so we always get a tuple, sometimes with an empty dictionary for the test_context
-    def wrapper(*args, **kwargs):
-        result = launch_description_fn(*args, **kwargs)
-        if isinstance(result, tuple):
-            return result
-        else:
-            return result, {}
+class _LaunchDiedException(Exception):
```

Comment:
The launch can die out from under you because of an error in the launch description, because all of the process accidentally died, or because the user hit ctrl+c.  With the ApexRunner split into `_RunnerWorker` and `ApexRunner` I needed a way to communicate up to the `ApexRunner` that this happened.